### PR TITLE
[TBDGen] Remove InstallAPI Flag logic

### PR DESCRIPTION
**Pull request by cyndyishida**
The linker & TAPI doesn't use it anymore.

We should likely remove the command line options for this too but that might need to be staged, this unblocks downstream CI for now.

### CODE REVIEW

**Comment by cyndyishida** [comment]
@swift-ci please test

File: test/TBD/installapi-flag.swift [hunk 1]
```diff
@@ -1,18 +0,0 @@
-// REQUIRES: VENDOR=apple 
-// RUN: %empty-directory(%t)
-
-// 1. Emit two TBDs, one with -tbd-is-installapi set, and one without
-
-// RUN: %target-swift-frontend -typecheck %s -tbd-is-installapi -emit-tbd -emit-tbd-path %t/flag-provided.tbd
```
Comment (tshortli) [inline test/TBD/installapi-flag.swift:6]: Should the `-tbd-is-installapi` flag also be deprecated and eventually removed?

**Review by tshortli — APPROVED** [review]

File: test/TBD/installapi-flag.swift (hunk 1 above)
Comment (cyndyishida) [reply test/TBD/installapi-flag.swift:6]: Yes. 
> We should likely remove the command line options for this too but that might need to be staged.
Upstream now has support for tbd-v5 which is in json but also holds more attributes that need to be recorded.
For swift, I think it's just capturing the segment of which symbols will be defined in. 
 I'll do more thorough update then.

**Comment by cyndyishida** [comment]
macOS test failures look fixed by: https://github.com/apple/swift/pull/64077 
I'll retry.

**Comment by cyndyishida** [comment]
@swift-ci please test

**Pull request merged**